authors/apps/analytics/views.py

```diff
@@ -0,0 +1,76 @@
+from rest_framework import status
+from rest_framework.permissions import (IsAuthenticated, )
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from authors.apps.analytics.models import ReadsReport
+from authors.apps.analytics.renderers import AnalyticsJSONRenderer
+from authors.apps.analytics.response_messages import REPORT_MSG
+from authors.apps.analytics.serializers import ReportAPISerializer
+from authors.apps.articles.models import Articles
+from authors.apps.articles.permissions import IsVerified
+
+
+class AnalyticsReportAPIView(APIView):
+    """
+    Handles viewing of read article analytics
+    Traffic and visitor statistics are available for stories if authenticated
+    """
+    permission_classes = (IsAuthenticated, IsVerified,)
+    serializer_class = ReportAPISerializer
+    renderer_classes = (AnalyticsJSONRenderer,)
+
+    def get(self, request):
+        """
+        Handles listing all report on an article
+
+        :param request:
+        :return: [report]
+        """
+        self.check_permissions(request)
+
+        report = ReadsReport.objects.filter(user=request.user)
+
+        serializer = self.serializer_class(report, many=True)
+
+        return Response(serializer.data,
+                        status=status.HTTP_200_OK)
+
+
+class AnalyticsUpdateReportAPIView(APIView):
+    """
+    Handles updating viewing of read article analytics
+    Traffic and visitor statistics are available for stories if authenticated
+    """
+    permission_classes = (IsAuthenticated, IsVerified,)
+    serializer_class = ReportAPISerializer
+    renderer_classes = (AnalyticsJSONRenderer,)
+
+    def patch(self, request, slug):
+        """
+        Handles updating report if author is same as the requester and article is similar
+        :param request:
+        :param slug:
+        :return: [updated report]
+        """
+
+        self.check_permissions(request)
+
+        try:
+            article = Articles.objects.get(slug=slug)
+            report = ReadsReport.objects.get(user=request.user, article=article)
+            if report.full_read is True:
+                return Response({
+                                    "errors": REPORT_MSG['REPORT_UPTO_DATE']},
+                                status=status.HTTP_400_BAD_REQUEST)
+            if request.data['full_read'] != report.full_read:
+                serializer = self.serializer_class(report, data=request.data, partial=True)
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            return Response({
+                                "errors": REPORT_MSG['CAN_NOT_UNREAD']}, status=status.HTTP_400_BAD_REQUEST)
+        except Articles.DoesNotExist:
```

Comment:
This exception handler misses the case where an instance of `ReadsReport` is not found. Consider updating the exception handler to catch that as well.

Here's a sample screenshot:
<img width="557" alt="Screenshot 2019-04-24 at 11 06 08" src="https://user-images.githubusercontent.com/30801697/56643166-c006db80-6681-11e9-8227-f80f0af530b3.png">

Reply:
Done. Kindly review once again.